presets/ragengine/main.py

```diff
@@ -93,14 +129,24 @@ def health_check():
     """,
 )
 async def index_documents(request: IndexRequest):
+    # Use a context manager for measuring the time
+    start_time = time.time()
     try:
         doc_ids = await rag_ops.index(request.index_name, request.documents)
         documents = [
             DocumentResponse(doc_id=doc_id, text=doc.text, metadata=doc.metadata)
             for doc_id, doc in zip(doc_ids, request.documents)
         ]
+        # Record successful request
+        rag_index_requests_total.labels(status=STATUS_SUCCESS).inc()
+        # Record latency
+        rag_index_latency.labels(status=STATUS_SUCCESS).observe(time.time() - start_time)
         return documents
     except Exception as e:
+        # Record failed request
+        rag_index_requests_total.labels(status=STATUS_FAILURE).inc()
+        # Record latency even for failures
+        rag_index_latency.labels(status=STATUS_FAILURE).observe(time.time() - start_time)
         raise HTTPException(status_code=500, detail=str(e))
```

Comment:
style choice but can use the try-except-finally here like you did above

```suggestion
async def index_documents(request: IndexRequest):
    # Use a context manager for measuring the time
    start_time = time.time()
    status = STATUS_FAILURE  # set default
    try:
        doc_ids = await rag_ops.index(request.index_name, request.documents)
        documents = [
            DocumentResponse(doc_id=doc_id, text=doc.text, metadata=doc.metadata)
            for doc_id, doc in zip(doc_ids, request.documents)
        ]
        # Record latency
        rag_index_requests_total.labels(status=STATUS_SUCCESS).inc()
        status = STATUS_SUCCESS
        return documents
    except Exception as e:
        rag_index_requests_total.labels(status=STATUS_FAILURE).inc()
        raise HTTPException(status_code=500, detail=str(e))
    finally:
        rag_index_latency.labels(status=status).observe(time.time() - start_time)
```

Reply:
same at other places